File: 240403/개인정보/personal-info.java
import java.util.*;

class Person implements Comparable<Person> {
    String name;
    int height;
    double weight;

    public Person(String name, int height, double weight) {
        this.name = name;
        this.height = height;
        this.weight = weight;
    }

    public void printInfo() {
        System.out.printf("%s %d %.1f", name, height, weight);
    }

    @Override
    public int compareTo(Person person) {
        return this.name.compareTo(person.name);
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Person[] p = new Person[5];

        for(int i = 0; i < 5; i ++) {
            p[i] = new Person(sc.next(), sc.nextInt(), sc.nextDouble());
        }

        Arrays.sort(p);

        System.out.println("name");
        for(int i = 0; i < 5; i ++) {
            p[i].printInfo();
            System.out.println();
        }

        System.out.println();

        Arrays.sort(p, new Comparator<Person>(){
            @Override
            public int compare(Person p1, Person p2){
                return p2.height - p1.height;
            }
        });

        System.out.println("height");
        for(int i = 0; i < 5; i ++) {
            p[i].printInfo();
            System.out.println();
        }
    }
}

File: 240616/개인정보/personal-info.py
class Person:
    def __init__(self, name, height, weight):
        self.name = name
        self.height = height
        self.weight = weight
    
    def __repr__(self):
        return f'{self.name} {self.height} {self.weight}'

_people = []

for i in range(5):
    p = input().split()
    new_person = Person(p[0], int(p[1]), float(p[2]))
    _people.append(new_person)

_people.sort(key=lambda x: x.name)

print('name')
for i in range(5):
    print(_people[i])

print()

_people.sort(key=lambda x: -x.height)

print('height')
for i in range(5):
    print(_people[i])